callout/src/test/java/com/google/apigee/ProtoMessageBuilders/ExecutionContextProtoMessageBuilderTest.java

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.apigee.ProtoMessageBuilders;
+
+import com.apigee.flow.Fault;
+import com.apigee.flow.execution.ExecutionContext;
+import com.google.apigee.Execute;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+public class ExecutionContextProtoMessageBuilderTest {
+
+  private final String ATTRIBUTES_KEY1 = "key1";
+  private final String ATTRIBUTES_KEY2 = "key2";
+  private final String ATTRIBUTES_VALUE1 = "value1";
+  private final String ATTRIBUTES_VALUE2 = "value2";
+  private final String FAULT_NAME = "faultname";
+  private final String FAULT_REASON = "faultreason";
+  private final String FAULT_SUB_CATEGORY = "faultsubcategory";
+  Fault mockFault1 = mock(Fault.class);
+  Fault mockFault2 = mock(Fault.class);
+  @Spy private FakeExecutionContext executionContext;
+
+  @Before
+  public void init() {
+    MockitoAnnotations.openMocks(this);
+
+    doReturn(Fault.Category.Messaging).when(mockFault1).getCategory();
```

Comment:
Could you make a fake implementation for `Fault`? All the `when` statements are hard to parse at first glance.

You might want to look into [AutoValue](https://github.com/google/auto/blob/master/value/userguide/index.md) as well.

Reply:
I don't think(?) I can make a fake implementation for Fault since it's not an interface and extending it would require calling the superclass constructor where is where our issue is. I did play around with AutoValue, but I'm not sure I can use it here for the same reason.

Reply:
sg, feel free to leave in the two faults and the when statements if you feel like it will help test your code better.